src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp

```diff
@@ -104,49 +124,149 @@ XBU::verbose("SubCommand option: read mem");
     throw xrt_core::error(std::errc::operation_canceled);
   }
 
+  // Validate the given address option
+  uint64_t addr = 0;
   try {
     //-- base address
-    addr = std::stoll(m_baseAddress, nullptr, 0);
+    addr = std::stoull(m_baseAddress, nullptr, 0);
   }
-  catch(const std::invalid_argument&) {
+  catch (const std::invalid_argument&) {
     std::cerr << boost::format("ERROR: '%s' is an invalid argument for '--address'\n") % m_baseAddress;
     throw xrt_core::error(std::errc::operation_canceled);
   }
 
+  // Validate the number of bytes to be written if defined
+  // This does not need to be defined for the --input option path
+  uint64_t size = 0;
   try {
-    //-- size
-    size = std::stoll(m_sizeBytes, nullptr, 0);
+    if (!m_sizeBytes.empty())
+      size = XBUtilities::string_to_base_units(m_sizeBytes, XBUtilities::unit::bytes);
   }
-  catch(const std::invalid_argument&) {
-    std::cerr << boost::format("ERROR: '%s' is an invalid argument for '--size'\n") % m_sizeBytes;
+  catch (const xrt_core::error& e) {
+    std::cerr << boost::format("Value supplied to --size is invalid: %s\n") % e.what();
     throw xrt_core::error(std::errc::operation_canceled);
   }
 
+  if (size == 0) {
+    std::cerr << boost::format("ERROR: Value for --size must be greater than 0\n");
+    throw xrt_core::error(std::errc::operation_canceled);
```

Comment:
Please look at ALL throw operations to see if the error message can be part of it.